### ISSUE TITLE
Extra group by in .count query 

### ISSUE THREAD

**Issue by artworx** [amberframework/granite#334]
This count query: 

```crystal
Wall.where(user_id: user_id).count
```

generates the sql: 

```sql
SELECT COUNT(*) FROM wall WHERE user_id = $1 GROUP BY user_id
```

I would expect the sql to not contain any `group by` clauses. 

The model looks like this: 

```crystal
class Wall < Granite::Base
   adapter pg
   table_name wall

   field user_id : Int64
   field post_id : Int64
   field following_id : Int64

   belongs_to :post

   field published_at : Time
end
```

**Comment by Blacksmoke16** [comment]
This is because aggregates require a group_by, but `count` does not.

**Comment by unequaled86** [comment]
fixed #352
